src/ContractInteract/EIP20Token.js

```diff
@@ -0,0 +1,179 @@
+// Copyright 2019 OpenST Ltd.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// ----------------------------------------------------------------------------
+//
+// http://www.simpletoken.org/
+//
+// ----------------------------------------------------------------------------
+
+'use strict';
+
+const BN = require('bn.js');
+const Web3 = require('web3');
+const Contracts = require('../Contracts');
+const Utils = require('../../src/utils/Utils');
+
+/**
+ * Contract interact for EIP20Token contract.
+ */
+class EIP20Token {
+  /**
+   * Constructor for EIP20Gateway.
+   *
+   * @param {Object} web3 Web3 object.
+   * @param {string} tokenAddress EIP20Token contract address.
+   */
+  constructor(web3, tokenAddress) {
+    if (web3 instanceof Web3) {
+      this.web3 = web3;
+    } else {
+      const err = new TypeError(
+        "Mandatory Parameter 'web3' is missing or invalid",
+      );
+      throw err;
+    }
+
+    if (!Web3.utils.isAddress(tokenAddress)) {
+      const err = new TypeError(
+        "Mandatory Parameter 'tokenAddress' is missing or invalid.",
+      );
+      throw err;
+    }
+
+    this.tokenAddress = tokenAddress;
+
+    this.contract = Contracts.getEIP20Token(this.web3, this.tokenAddress);
+
+    if (!this.contract) {
+      const err = new TypeError(
+        `Could not load token contract for: ${this.tokenAddress}`,
+      );
+      throw err;
+    }
+
+    this.approve = this.approve.bind(this);
+    this._approveRawTx = this._approveRawTx.bind(this);
+    this.allowance = this.allowance.bind(this);
+  }
+
+  /**
+   * Approves account address for the amount transfer.
+   *
+   * @param {string} spenderAddress Spender account address.
+   * @param {string} amount Approve amount.
+   * @param {string} txOptions Transaction options.
+   *
+   * @returns {Promise} Promise object.
+   */
+  approve(spenderAddress, amount, txOptions) {
+    return new Promise((onResolve, onReject) => {
+      if (!txOptions) {
+        const err = new TypeError('Invalid transaction options.');
+        onReject(err);
```

Comment:
When resolving, rejecting a promise in the middle of a function, it should return early, to prevent unexpected sideeffects:
```
return onReject(err);
```